Tests/SwiftParserTest/IncrementalParsingTests.swift

```diff
@@ -13,34 +13,43 @@
 import XCTest
 import SwiftSyntax
 import SwiftParser
+import _SwiftSyntaxTestSupport
 
 public class IncrementalParsingTests: XCTestCase {
-
   public func testIncrementalInvalid() {
     let original = "struct A { func f() {"
-    let step: (String, (Int, Int, String)) =
-      ("struct AA { func f() {", (8, 0, "A"))
-
-    var tree = Parser.parse(source: original)
-    let sourceEdit = SourceEdit(range: ByteSourceRange(offset: step.1.0, length: step.1.1), replacementLength: step.1.2.utf8.count)
-    let lookup = IncrementalParseTransition(previousTree: tree, edits: ConcurrentEdits(sourceEdit))
-    tree = Parser.parse(source: step.0, parseTransition: lookup)
-    XCTAssertEqual("\(tree)", step.0)
+    let newSource = "struct AA { func f() {"
+
+    let concurrentEdits = getConcurrentEdits(from: original, to: newSource)
+
+    let oldTree = Parser.parse(source: original)
+    let lookup = IncrementalParseTransition(previousTree: oldTree, edits: concurrentEdits)
+    let newTree = Parser.parse(source: newSource, parseTransition: lookup)
+
+    XCTAssertEqual("\(newTree)", newSource)
   }
 
   public func testReusedNode() throws {
     try XCTSkipIf(true, "Swift parser does not handle node reuse yet")
 
-    let original = "struct A {}\nstruct B {}\n"
-    let step: (String, (Int, Int, String)) =
-      ("struct AA {}\nstruct B {}\n", (8, 0, "A"))
+    let original =
+      """
+      struct A {}
+      struct B {}
+      """
+
+    let newSource =
+      """
+      struct A {}
```

Comment:
`newSource` should have `struct AA` instead of `struct A`

```suggestion
      struct AA {}
```